docs/src/components/PageHeader.js

```diff
@@ -31,17 +41,59 @@ export default function ComponentHeader({
   fileName,
   showSourceLink = true,
 }: Props): Node {
+  const { pinnedSection, setPinnedSection } = useNavigationSidebarContext();
+
+  function handlePinnedComponents() {
+    const arrayPinned = JSON.parse(pinnedSection);
+    if (!arrayPinned.includes(fileName || name) && arrayPinned.length < 3) {
+      return setPinnedSection(
+        JSON.stringify([...arrayPinned, fileName || name])
+      );
+    }
+    return setPinnedSection(
+      JSON.stringify(arrayPinned.filter((item) => item !== (fileName || name)))
+    );
+  }
+
+  const pinnedSectionArr = JSON.parse(pinnedSection);
+  const pinnedSectionMax = pinnedSectionArr.length < 3;
```

Comment:
@AlbertCarreras might be good to have the max number of pinned items in a const on the top of the file.

Reply:
Oh yeah that makes size

Reply:
Sense